# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/AnushkaWagh03/Flyrank-ML-IIntern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I am choosing the **Refresh / Content Opportunity Scoring** lane. The decision I want to support is which existing content items should be reviewed first for possible refresh. This lane fits the starter data because it contains page-level search and engagement signals such as impressions, clicks, sessions, CTR, position, content age, and trend measurements. The starter dataset has 30,000 content items, so there is enough breadth to investigate whether several observable signals can produce a useful review priority. I am treating the starter `trend_direction == down` label as a current-window proxy, not as proof of future decline; over the next seven weeks I want to test a future-looking outcome where possible.

In [3]:
import pandas as pd

path = "../../data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(path)

down_count = df["trend_direction"].astype(str).str.lower().eq("down").sum()

print(f"Starter rows: {len(df)}")
print(f"Columns: {df.shape[1]}")
print(f"Rows with trend_direction = down: {down_count}")
print(f"Share currently labeled down: {down_count / len(df):.2%}")

Starter rows: 30000
Columns: 44
Rows with trend_direction = down: 16262
Share currently labeled down: 54.21%


## 2. The question: decision, action, cost of a wrong call

**Research question:** Among existing content items, which pages should an editor review first because their observed search/engagement signals suggest a refresh opportunity?

**Decision:** Decide which pages enter the refresh-review queue and how the queue should be ordered.

**Action:** A content/editorial reviewer checks the highest-ranked pages, verifies the evidence and page context, and decides whether to refresh, investigate CTR/engagement, or monitor the page. The model or score is a reviewer aid, not an automatic publishing decision.

**Output:** A ranked opportunity score/queue with an interpretable reason for why an item was prioritized.

**Cost of a wrong call:** A false positive spends limited editorial time reviewing a page that does not need work. A false negative can delay attention on a genuinely deteriorating or valuable page. Therefore, ranking quality near the top of the queue matters more than making every item perfectly classified.

**Why data/ML can help:** The dataset contains multiple observable signals that may interact: demand, clicks, sessions, CTR, position, age, content size, and recent activity. A simple rule is an important baseline, but if the useful pattern is distributed across several signals, a scoring model may rank candidates more consistently. I will only keep ML if it demonstrates value over a simple baseline using a pre-declared ranking metric such as precision@K.

In [4]:
print("Decision grain: one pseudonymized content item")
print("Candidate ranking metric: precision@K")

Decision grain: one pseudonymized content item
Candidate ranking metric: precision@K


## 3. Quick look at the data (2-3 real numbers)

The starter CSV contains **30,000 rows across 44 columns**. **16,262 rows (54.21%)** currently have `trend_direction = down`. These numbers make the refresh lane worth investigating because the starter set is large enough for comparisons and the current-window proxy identifies a substantial group of items showing a downward trend. However, the 54.21% figure is not a future decline rate: the starter label is derived from the current trend measurement. The next step is therefore to test whether observable signals can support a better, future-looking review decision rather than simply reproducing that label.

In [5]:
# Re-check the supporting numbers directly from the starter data.

assert len(df) == 30000
assert down_count == 16262

print(f"Rows: {len(df)}")
print(f"Rows currently down: {down_count}")
print(f"Current down-label share: {down_count / len(df):.2%}")

Rows: 30000
Rows currently down: 16262
Current down-label share: 54.21%


## 4. Careful words: what I can and can't claim

**I can claim:** I can describe observed patterns in the starter data and, after proper validation, report whether a scoring approach provides useful **directional decision-support** for prioritizing review. I can compare a simple baseline with a model using a ranking metric and a leakage-aware validation design.

**I cannot claim:** I cannot claim that the score causes traffic changes, that a page will definitely decline, that refreshing a page will increase traffic, or that the model predicts Google's ranking decisions. I also cannot treat `trend_direction` or `trend_pct` as independent predictive features when the starter label is derived from them.

The starter `is_declining_label` is a proxy because it is derived from the current trend window. A stronger capstone target should use an observed outcome from a later time window, for example features from an earlier window followed by measured decline or recovery over the next window. Any model will therefore be presented as evidence for prioritization, not as ground truth or an automatic editorial decision.

In [6]:
print("Leakage check: trend_direction is not used as a model feature.")
print("Starter label is a proxy derived from the current trend window.")

Leakage check: trend_direction is not used as a model feature.
Starter label is a proxy derived from the current trend window.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.